File: solutions/7kyu/Unique_string_characters/unique_string_characters.js
function solve(a,b){
  let filteredA = a.split('').filter(
      function(e) {
        return this.indexOf(e) < 0;
      },
      b.split('')
  );

  let filteredB = b.split('').filter(
      function(e) {
        return this.indexOf(e) < 0;
      },
      a.split('')
  );
  return filteredA.join('') + filteredB.join('');
}

console.log(solve("xyab", "xzca")); //ybzc"
console.log(solve("xyabb", "xzca")); //"ybbzc"

File: solutions/7kyu/Unique_string_characters/unique_string_characters.rb
def solve(a, b)
  ((a.split('').delete_if { |v| v =~ /["#{b}"]/ }) + (b.split('').delete_if { |v| v =~ /["#{a}"]/ })).join('')
end

p solve("xyab", "xzca") #ybzc"
p solve("xyabb", "xzca") #"ybbzc"
